packages/feature-toggles/src/middleware.ts

```diff
@@ -0,0 +1,43 @@
+import express from 'express'
+
+import { FeatureReceiver } from '.'
+import { FeatureState, isFeatureEnabled } from './universal'
+
+export interface WithFeatures {
+    features: FeatureState
+}
+
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Express {
+        // eslint-disable-next-line @typescript-eslint/no-empty-interface
+        export interface Request extends WithFeatures {}
+    }
+}
+
+export function createFeatureStateMiddleware(featureReceiver: FeatureReceiver) {
+    return feaureStateMiddleware
+
+    function feaureStateMiddleware(
+        req: express.Request & WithFeatures,
+        _res: express.Response,
+        next: express.NextFunction,
+    ) {
+        req.features = featureReceiver.featureState
+
+        if (next) {
```

Comment:
the type doesn't look optional, or is `express.NextFunction` optional in some way?

Reply:
Agreed!